packages/next/src/build/utils.ts

```diff
@@ -1479,6 +1479,19 @@ export async function isPageStatic({
           },
           {}
         )
+        if (!appConfig.dynamic) {
+          appConfig.dynamic = 'auto'
+        }
+
+        if (process.env.__NEXT_CONFIG_OUTPUT === 'export') {
```

Comment:
This file isn't compiled so it's likely that this needs to read from config instead?

Reply:
You're right! Fixed in [4915943](https://github.com/vercel/next.js/pull/47022/commits/491594361389823c79a4c7ba051fb67cef02fd6c)